### PR TITLE
Improvements to Python | Casting

### PR DESCRIPTION
**Pull request by kaifee-haque**
### Description

Make the following changes to `casting.md`:
- Add tags
- Fix typos/grammar
- Remove some example sections for brevity
- Add section on a use case for casting

### Type of Change

- Editing an existing entry (fixing a typo, bug, issues, etc)

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.

### CODE REVIEW

File: content/python/concepts/casting/casting.md
Comment (yangc95) [inline content/python/concepts/casting/casting.md:16]:
```suggestion
Casting, also known as type conversion, is a process that converts a variable's data type into another data type. These conversions can be implicit (automatically interpreted) or explicit (using built-in functions).
```

File: content/python/concepts/casting/casting.md
Comment (yangc95) [inline content/python/concepts/casting/casting.md:22]:
```suggestion
In order to avoid data loss, Python converts smaller data types to higher data types. In the below example, the integer `7` is type casted to a `float` when added with `2.2`. Since the expression represents the sum of two `float` values, the data type of `y` is also `float`.
```

File: content/python/concepts/casting/casting.md
Comment (yangc95) [inline content/python/concepts/casting/casting.md:40]: Wanting to confirm, are we speaking about conversion constructors in this second sentence? Would you mind linking me to your source? Thanks in advance! ☺️

File: content/python/concepts/casting/casting.md
Comment (yangc95) [inline content/python/concepts/casting/casting.md:98]:
```suggestion
In the above example, `x` must be the string representation of an integer for casting to work. `x` is a string that contains a number so there's no error when type casting.
```

File: content/python/concepts/casting/casting.md
Comment (yangc95) [inline content/python/concepts/casting/casting.md:117]:
```suggestion
When operating on data, it is important to keep their type in mind. The following codebyte is a flawed attempt to print the square of a number specified by the user. Try running it and ponder the result.
```

File: content/python/concepts/casting/casting.md
Comment (yangc95) [inline content/python/concepts/casting/casting.md:124]:
```suggestion
The `input()` function takes input from the user and stores it in a variable as a string. However, the `**` operator takes two numbers and returns the first number to the power of the second. In order to make the code work, the input variable must be cast to a number type. Try to edit the above Codebyte to successfully square a number inputted by the user.
```

File: content/python/concepts/casting/casting.md
Comment (yangc95) [reply content/python/concepts/casting/casting.md:117]: I don't believe we want to place flawed codebytes in Docs. @sonnynomnom can this be confirmed?

Rather, we can condense all the deleted code snippets and add a more complex example in there as well.

**Review by yangc95 — COMMENTED** [review]
Nice to see a familiar user! @kaifee-haque I like what you've done with these enhancements.

After everything is resolved with the comments/suggestions, I'll take another look and hopefully approve this PR on my end. Make sure to tag me 🏷️ and let me know if you have any questions!

File: content/python/concepts/casting/casting.md
Comment (hughlilly) [reply content/python/concepts/casting/casting.md:22]: Sorry to butt in on someone else's work here, but I believe the past tense of "cast" is "cast" — as with "broadcast" or when talking about an actor having been "cast" in a play or movie, for example. Therefore I think this should read:

>the integer `7` is type cast to a `float` when added with `2.2`.

**Comment by yangc95** [comment]
Hey @kaifee-haque! 🎄 

How's it going with this PR? Please reach out if you have any questions. Also, let me know if you'd like me to make direct changes based on my/others' recommendations.

**Review by yangc95 — APPROVED** [review]
Ready for review 2️⃣ 👍

**Comment by KTom101** [comment]
Hi @yangc95, I'll start on the second review!

File: content/python/concepts/casting/casting.md
Comment (KTom101) [inline content/python/concepts/casting/casting.md:22]:
```suggestion
To avoid data loss, Python converts smaller data types to higher data types. In the below example, the integer 7 is converted to a float when added with 2.2. Since the expression represents the sum of two `float` values, the data type of `y` is also `float`.
```

File: content/python/concepts/casting/casting.md
Comment (KTom101) [inline content/python/concepts/casting/casting.md:117]:
```suggestion
When operating on data, it is important to keep their type in mind. The following code is a flawed attempt to print the square of a number specified by the user. Try running it and ponder the result.
```

File: content/python/concepts/casting/casting.md
Comment (KTom101) [inline content/python/concepts/casting/casting.md:124]:
```suggestion
The `input()` function takes input from the user and stores it in a variable as a string. However, the `**` operator takes two numbers and returns the first number to the power of the second. In order to make the code work, the input variable must be cast to a number type. Try to edit the above code to successfully square a number inputted by the user.
```

Comment (KTom101) [inline content/python/concepts/casting/casting.md:124]:
```suggestion
The `input()` function takes input from the user and stores it in a variable as a string. However, the `**` operator takes two numbers and returns the first number to the power of the second. In order to make the code work, the input variable must be cast to a number type. Try to edit the code above to successfully square a number inputted by the user.
```

File: content/python/concepts/casting/casting.md
Comment (KTom101) [inline content/python/concepts/casting/casting.md:117]:
```suggestion
When operating on data, it is important to keep their data type in mind. The following code is a flawed attempt to print the square of a number specified by the user. Try running it and ponder the result.
```

**Review by KTom101 — APPROVED** [review]
Changed some minor things.

**Comment by Dusch4593** [comment]
Hi @kaifee-haque! I'm going to take one last look at this PR before we merge! 😄

File: content/python/concepts/casting/casting.md
Comment (Dusch4593) [inline content/python/concepts/casting/casting.md:16]: I like this new-and-improved first sentence! Could we include this in the meta-`Description` above on line 3?

File: content/python/concepts/casting/casting.md
Comment (Dusch4593) [inline content/python/concepts/casting/casting.md:20]:
```suggestion
Python's interpreter automatically performs type conversion without user involvement.
```

File: content/python/concepts/casting/casting.md
Comment (Dusch4593) [inline content/python/concepts/casting/casting.md:28]:
```suggestion
print(y) 
# Output: 9.2

print(type(y))
# Output: <class 'float'>
```

If you accept these recommended changes, I also (highly) recommend deleting the content in lines 31-36 altogether.

File: content/python/concepts/casting/casting.md
Comment (Dusch4593) [inline content/python/concepts/casting/casting.md:22]:
```suggestion
Python avoids data loss by converting lower data types to higher data types. In the example below, the integer, 7, is converted to a float when added with another float, 2.2. .
```

> Python converts smaller data types to higher data types.

What do we mean here when referring to some Python data types as "smaller/lower" and others as "higher"?

File: content/python/concepts/casting/casting.md
Comment (Dusch4593) [inline content/python/concepts/casting/casting.md:25]:
```suggestion
y = 7 + 2.2 
# Python automatically type casts y into float
```

File: content/python/concepts/casting/casting.md
Comment (Dusch4593) [inline content/python/concepts/casting/casting.md:42]:
```suggestion
- The `str()` function takes an integer or float as an argument and converts it to a string.
```

File: content/python/concepts/casting/casting.md
Comment (Dusch4593) [inline content/python/concepts/casting/casting.md:40]:
```suggestion
Explicit type casting involves Python's predefined functions, which act as a constructor of another data type:
```

For

File: content/python/concepts/casting/casting.md
Comment (Dusch4593) [inline content/python/concepts/casting/casting.md:44]:
```suggestion
- The `str()` function takes an integer or float as an argument and converts it to a string.
- The [`int()` function](https://www.codecademy.com/resources/docs/python/built-in-functions/int) takes a string or float as an argument converts it to an integer.
- The [`float()` function](https://www.codecademy.com/resources/docs/python/built-in-functions/float) takes an integer or string as an argument and converts it to a float.
```

File: content/python/concepts/casting/casting.md
Comment (Dusch4593) [inline content/python/concepts/casting/casting.md:46]: Given I recently filed [a related issue (#486) about the `str()` function](https://github.com/Codecademy/docs/issues/486), I suggest omitting lines 46-114. Perhaps after this PR is merged, you would be interested in being assigned to write the term entry for the `str()` Python function? @kaifee-haque

File: content/python/concepts/casting/casting.md
Comment (Dusch4593) [inline content/python/concepts/casting/casting.md:117]:
```suggestion
When operating on data, it is important to be mindful of the data types associated with it. The following code is a flawed attempt to print the square of a number specified by the user. When run, a `TypeError` will be returned:
```

File: content/python/concepts/casting/casting.md
Comment (Dusch4593) [inline content/python/concepts/casting/casting.md:124]:
```suggestion
The [`input()` function](https://www.codecademy.com/resources/docs/python/built-in-functions/input) takes input from the user and stores it in a variable as a string. However, the `**` operator takes two numbers and returns the first number to the power of the second. In order to make the code work, the input variable must be cast to a number type. Try to edit the code above to successfully square a number inputted by the user.
```

**Review by Dusch4593 — CHANGES_REQUESTED** [review]
@kaifee-haque All finished with my review! Take a look at my comments/suggestions and tag us when you've finished changes. 😃

**Comment by yangc95** [comment]
> @kaifee-haque All finished with my review! Take a look at my comments/suggestions and tag us when you've finished changes. 😃

Since it's been some time, I'm going to go in and make changes 👍  @Dusch4593 @kaifee-haque

File: content/python/concepts/casting/casting.md
Comment (yangc95) [reply content/python/concepts/casting/casting.md]: omitted those lines as suggested! @Dusch4593 @kaifee-haque

File: content/python/concepts/casting/casting.md
Comment (yangc95) [reply content/python/concepts/casting/casting.md:22]: @Dusch4593 to my understanding, a lower data type refers to one that allows for a smaller range of values whereas a high data type allows for more range. So, integers are constrained to represent whole numbers while floats can represent whole and decimal numbers.

**Review by Dusch4593 — APPROVED** [review]
LGTM! 👍

**Comment by yangc95** [comment]
@kaifee-haque, your entry is live here!
https://www.codecademy.com/resources/docs/python/casting

Thank you for this enhancement! Also thanks to @hughlilly for your input as well ☺️ 

cc: @KTom101 @Dusch4593

**Pull request merged**
